sys/net/application_layer/dns/dns.c

```diff
@@ -216,14 +188,43 @@ int sock_dns_query(const char *domain_name, void *addr_out, int family)
     }
 
     for (int i = 0; i < SOCK_DNS_RETRIES; i++) {
+        uint8_t *buf = dns_buf;
+
+        sock_dns_hdr_t *hdr = (sock_dns_hdr_t*) buf;
+        memset(hdr, 0, sizeof(*hdr));
+        hdr->id++; /* random? */
```

Comment:
This wasn't what I was asking of you... This is not correct AFAIK.

1. The `id` should not increase with every retry, but every transaction.
2. It is now just 1 every time...

What I actually wanted is
```suggestion
        hdr->id = id;
```

with `id` being declared as

```C
uint16_t id = 0; /* random? */
```

outside of the for-loops scope. This way, you

1. don't change behavior
2. allow for future devs that do want to introduce requested randomness to not fall into the same trap you just did ;-).